app/controllers/api/base_controller/normalizer.rb

```diff
@@ -20,37 +17,19 @@ def normalize_hash(type, obj, opts = {})
         end
 
         attrs.each do |k|
-          value =  normalize_direct(type, k, obj.kind_of?(ActiveRecord::Base) ? obj.try(k) : obj[k])
+          value = obj.kind_of?(ActiveRecord::Base) ? normalize_attr(k, obj.try(k)) : normalize_attr(k, obj[k])
```

Comment:
Just wondering, as a further refactoring: do we need to switch here? AR objects have the same interface:

```
u = User.first
=> #<User id: 1, name: "Administrator", email: nil, icon: nil, created_on: "2016-05-02 17:24:56", updated_on: "2017-04-28 17:28:39", userid: "admin", settings: {}, lastlogon: "2017-04-28 17:28:39", lastlogoff: "2017-04-26 21:21:37", current_group_id: 2, first_name: nil, last_name: nil, password_digest: "$2a$10$mlbjgYkN13sa59UjfjbS4u8FWIcjGwmzyRQXZPVcayf...">
irb(main):002:0> u.name
=> "Administrator"
irb(main):003:0> u["name"]
=> "Administrator"
irb(main):004:0> u["foo"]
=> nil
```